x-pack/plugins/observability/public/components/shared/field_value_suggestions/field_value_selection.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FormEvent, Fragment, useEffect, useState, Dispatch, SetStateAction } from 'react';
+import {
+  EuiButton,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiPopoverTitle,
+  EuiSelectable,
+  EuiSelectableOption,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+export interface FieldValueSelectionProps {
+  value?: string;
+  label: string;
+  loading: boolean;
+  onChange: (val: string) => void;
+  values?: string[];
+  setQuery: Dispatch<SetStateAction<string>>;
+}
+
+const formatOptions = (values?: string[], value?: string): EuiSelectableOption[] => {
+  return (values ?? []).map((val) => ({ label: val, ...(value ? { checked: 'on' } : {}) }));
```

Comment:
Would
```
return (values ?? []).map((val) => ({ label: val, value: value ? { checked: 'on' } : { checked: 'off' }}));
```
work here. The spreading feels harder to read to me, but that's just a preference

Reply:
i think EuiSelectable only works with `checked`.